### ISSUE TITLE
2.2 Compatibility

### ISSUE THREAD

**Issue by Silarn** [amzn/amazon-payments-magento-2-plugin#133]
I know it isn't officially released yet, though they seem to be wrapping up development.

Of course the primary issue is the composer.json needs to be updated to reflect the changed framework/module versions and PHP version requirements. (Dropping 5.6 and adding 7.1). I feel like the other package requirements are a little extraneous and are fine to have simply in the etc/module.xml files.

I'm doing some tests in my forked repo and will report back when possible. Didn't run into any compile issues with Amazon's module, but I'll need to work through other module incompatibilities before I can fully debug any potential problems.

**Comment by danielneu** [comment]
@Silarn 

thanks a lot, very appreciated.
We are also working on this internally and will be able to incorporate any feedback you might have.

Best wishes 
Daniel

**Comment by andidhouse** [comment]
2.2 is released today so it would be really great to have a timeline when amazon pay (best also as a "normal payment option" for one step checkouts is released).

Also an info would be great if it takes some weeks as we need to plan this.

Thanks Daniel!

**Comment by alexander-sibert** [comment]
I removed Amazon Pay Extension for Upgrade to Magento 2.2. I hope you make soon your Extension compatible to Magento 2.2

**Comment by Silarn** [comment]
Still working through some other bugs before we can properly test functionality on 2.2, but with simple composer.json tweaks it installs and compiles without issue. I'll keep you posted. We've been working through some static content compile issues now that the compiles no longer fails gracefully...

**Comment by Silarn** [comment]
The checkout JS is not working properly with the Magento 2.2 JS bundler - but it seems to work fine with bundling disabled. At the moment not sure what the best solution for this is.

**Comment by Silarn** [comment]
In particular, it appears to happen (particularly if reloading the checkout page) here:

```
Uncaught TypeError: Cannot read property 'getElementsByTagName' of null
    at Object.c.Button (https://static-na.payments-amazon.com/OffAmazonPayments/us/sandbox/js/Widgets.js?nomin:43:143716)
```
src/Payment/view/frontend/web/js/amazon-button.js
```js
_renderAmazonButton: function() {
            var authRequest;
            OffAmazonPayments.Button($button.attr('id'), _this.options.merchantId, {
                type: _this.options.buttonType,
                color: _this.options.buttonColor,
                size: _this.options.buttonSize,
                language: _this.options.buttonLanguage,
                authorization: function() {
                    authRequest = amazon.Login.authorize(_this._getLoginOptions(), _this._popupCallback());
                }
            });
        },
```

I'm not quite sure what's trying to instantiate this as it isn't defined in the HTML - though the script is being loaded.

**Comment by danielneu** [comment]
@gewaechshaus 

I'm awaiting a PR for 2.2. compatibility any time. Once this is available I will update the master with it.
Changes seem to be minimal and only concentrating on the composer.json.

**Comment by danielneu** [comment]
The master should now have the compatibility in place, just merged the PR for it.

Best wishes 
Daniel

**Comment by gewaechshaus** [comment]
@danielneu - thx, that does work till now... Feel free to test at https://www.makerdise.com

I will put on logging for some time...

**Comment by danielneu** [comment]
Thanks Jan for the confirmation.

**Comment by Silarn** [comment]
We're wrapping up our M2.2 testing/fixes here and when we get the major bugs sorted out I may take another look at the checkout JS.

**Comment by danielneu** [comment]
General 2.2 compatibility is given, in case anything pops up, please create a new issue.
We are not seeing any JS errors like the one given above.

Closing this one.

Best wishes 
Daniel
